beangulp/testing.py

```diff
@@ -76,90 +122,72 @@ def log(msg, level=0, **kwargs):
 
 @click.command('test', short_help="Test the importer.")
 @click.argument('documents',type=click.Path(exists=True), nargs=-1)
-@click.option('--expected', '-e', type=click.Path(file_okay=False), metavar='DIR',
-              help="Directory containing the expecrted output files.")
 @click.option('--generate', '-g', is_flag=True, help="Generate the expected output files.")
 @click.option('--verbose', '-v', count=True, help="Enable verbose output.")
 @click.option('--quiet', '-q', count=True, help="Suppress all output.")
 @click.option('--exitfirst', '-x', is_flag=True, help="Stop at the first test failure.")
 @click.pass_obj
-def _test(ctx, documents, expected, generate, verbose, quiet, exitfirst):
+def _test(ctx, documents: List[str], generate: bool,
+          verbose: bool, quiet: bool, exitfirst: bool):
     """Run the importer on all DOCUMENTS and verify that it produces the
     desired output.  The desired output is stored in Beancount ledger
     files with an header containing additional metadata.  These files
     are named as the SHA1 digest of the documents and can be generated
     with the --generate option.
 
-    DOCUMENTS can be files or directories.  Directories are walked and
-    the importer is called on all files.  When no DOCUMENTS path is
-    specified on the command line, the tests/documents/ directory
-    relative to the test script location is used.  All documents for
-    which an expected output file exist must be positively identified
-    by the importer.
-
-    Expected output files are searched in the tests/expected/$NAME/
-    directory relative to the test script location, where $NAME is
-    replaced with the importer name, if the path exists, or in the
-    tests/expected/ directory.  This path can be overwritten through
-    the --expected DIR option.
-
+    DOCUMENTS can be files or directories. Directories are walked and the
+    importer is called on all files found, except for .beancount files.
     """
     assert len(ctx.importers) == 1
     importer = ctx.importers[0]
 
-    import __main__
-    root = os.path.dirname(__main__.__file__)
-
-    if not documents:
-        documents = (os.path.join(root, 'tests','documents'), )
-
-    if not expected:
-        expected = os.path.join(root, 'tests', 'expected', importer.name())
-        if not os.path.exists(expected):
-            expected = os.path.join(root, 'tests', 'expected')
-
```

Comment:
Removing this removes support for my use case: I have a few importers and I want to make sure that each identifies all the files it should identify, but not others. For ensuring this, I put all my test documents for all the importers in the same folder:
```
beans/
  ledger.beancount
  import.py
  importers/
     foo.py
     bar.py
     baz.py
     tests/
       documents/
         # all documents here
       expected/
         foo/
           # expected output for importer foo
         ...
```
In this way if and only if an expected output file exists in `expected/foo/` the foo importer should identify it. If you flatten the `exported/` directory hierarchy, I don't see an equivalently easy way to check this.

Reply:
Hmm, okay. Maybe we should provide an optional override for the directory name. I do think the default should be as straightforward as possible though - remember the audience.

Reply:
Done. I've reinserted the option, but it assume the same basenames as the original files.
The default remains to find the files in the same directory.
Good enough?

Reply:
I would prefer a scheme in which we look for the structured directories first and fall back to a flat directory structure if these are not there. Also, I don't think that having testing documents and expected output files in separate directory is such a cognitive load that it cannot be handled by someone expected to use this code.